### PR TITLE
Add a new principle: Code an Off Switch

### PR DESCRIPTION
**Pull request by kealanparr**
Hi Adam,

Liked the site enough that I thought I'd contribute.

Hope this finds you well, and is up to scratch

### CODE REVIEW

**Comment by AdamCraven** [comment]
Thank you Kealen for taking time to submit a principle. I'm away at the moment - but I will take a look tomorrow.

File: code-an-off-swtich.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+This is normally done via **Feature Flags** (AKA **Feature Toggles**)
+
+Giving yourself an easy way out if something goes wrong on site, pays you back tenfold for the time you do need it, alongside reducing risk on your site.
+
+It helps ensure:
+
+* You plan out your work in a modular way, to ensure your work can simply be "removed" and it falls back into an alternate state.
+* Provides a smooth release path for you to deploy your code.
+* Gives your on-call developers a very simple resolution if they get waken up at 1am to fix it.
```
Comment (AdamCraven) [inline code-an-off-swtich.md]: waken => woken

File: code-an-off-swtich.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
++++
+principle = "Code an off-switch"
+tags = ["feature flag", "feature toggle", "off switch"]
+summary = "Always plan for a way to switch your work off"
+authors = ["Kealan Parr"]
```
Comment (AdamCraven) [inline code-an-off-swtich.md]: You may want to change "Kealen Parr" to "kealanparr" (github username) if you want it to link to your github and principles.dev account - you should log in first to https://principles.dev/accounts/login/ using github. You will also get a founding badge on your profile for being one the very first people to submit a principle as well.

File: code-an-off-swtich.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+contributors = ["Kealan Parr"]
```
Comment (AdamCraven) [inline code-an-off-swtich.md]: You should remove yourself as a contributor, as you cannot be both a contributor and author

https://principles.dev/documentation/#contributors

Comment (AdamCraven) [reply code-an-off-swtich.md]:
```suggestion
contributors = []
```

File: code-an-off-swtich.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
++++
+principle = "Code an off-switch"
+tags = ["feature flag", "feature toggle", "off switch"]
```
Comment (AdamCraven) [inline code-an-off-swtich.md]: You can add tags: "practices", "team", "medium" for better visibility on-site

**Comment by AdamCraven** [comment]
Thank you for taking the time to improve everyone's capability, Kealan. 

The principle looks good and satisfies the requirement of "basic principle". Once you've made the changes, I'll merge it in to release branch and onto the principles.dev website.

Over-time you can carry on developing the principle further or you may find other people might also contribute to your principle to further develop it.

**Comment by kealanparr** [comment]
> Over-time you can carry on developing the principle further or you may find other people might also contribute to your principle to further develop it.

Sounds great, thanks for the feedback!

**Comment by kealanparr** [comment]
I think that's all the discussions resolved, and accordingly acted upon. Let me know if I missed anything 👍

**Comment by AdamCraven** [comment]
@kealanparr all merged in. It's now live on the site https://principles.dev/p/code-an-off-switch/ and linked together on your profile: https://principles.dev/u/kealanparr/

Great work and thanks for being a part of this project.

**Pull request merged**